### ISSUE TITLE
BelongsTo field not being sortable on Lens

### ISSUE THREAD

**Issue by skoyah** [laravel/nova-issues#2122]
- Laravel Version: 5.8
- Nova Version: 2.1.0
- PHP Version: 7.3

### Description
Having created a lens for a resource, and defined a belongsTo field, when sorting by that relationship, an error is thrown.

The model has a `province_id` field and a `province` relationship.

When sorting by this relationship, it tries to sort by the relationship name instead of the actual field.

![Screenshot 2019-11-08 at 16 16 22](https://user-images.githubusercontent.com/36083448/68487873-32b55e00-0244-11ea-99c0-8463a07f1de5.png)

**Comment by skoyah** [comment]
@jbrooksuk can you add the bug label?

**Comment by m-lotze** [comment]
BelongsTo is still not sortable at all. It only sorts by the ID column in the resource table. which is as useful as being not sortable.

there a multiple issues about it: https://github.com/laravel/nova-issues/issues/2124, https://github.com/laravel/nova-issues/issues/1821

two pull requests would have resolved it, but both got closed: https://github.com/laravel/nova/pull/726, https://github.com/laravel/nova/pull/619

maybe this time...

**Comment by skoyah** [comment]
@jbrooksuk do we have some feedback on this issue?

**Comment by janeannpaul** [comment]
Having the same issue, Is there any way to get it solved?

**Comment by mihai-scurtu** [comment]
I just came across this issue I believe this is the cause: the `Index` component orders using the `sortableUriKey` field while the `Lens` component wrongly uses the `attribute` field.

**Index.vue:**
```javascript
    orderByField(field) {
      let direction = this.currentOrderByDirection == 'asc' ? 'desc' : 'asc'

      if (this.currentOrderBy != field.sortableUriKey) {
        direction = 'asc'
      }

      this.updateQueryString({
        [this.orderByParameter]: field.sortableUriKey,
        [this.orderByDirectionParameter]: direction,
      })
    },
```

**Lens.vue**
```javascript
    orderByField(field) {
      var direction = this.currentOrderByDirection == 'asc' ? 'desc' : 'asc'
      if (this.currentOrderBy != field.attribute) {
        direction = 'asc'
      }
      this.updateQueryString({
        [this.orderByParameter]: field.attribute,
        [this.orderByDirectionParameter]: direction,
      })
    },
```
I would be happy to submit a PR if there isn't something I'm missing. I'm still getting to understand the nova codebase, so I'm not 100% this wouldn't break anything.

**Comment by orlyapps** [comment]
@mihai-scurtu that looks like the reason. Do you have made a PR?

**Comment by crynobone** [comment]
@orlyapps @mihai-scurtu 

Based on the latest code, the changes has been made to Nova since https://github.com/laravel/nova/commit/e2f5f303bffd8aff9e1c5fffe305ee6667e33e04 (available from v3.8.0 - v3.12.1). Can you guys verify if the issue still persist? 

I'll try to replicate the issue as well today.
